easy-ui-react/src/Button/Button.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactElement, ReactNode, useRef } from "react";
+import { IconProps } from "Icon";
+import { useButton } from "@react-aria/button";
+import { classNames, variationName } from "../utilities/css";
+
+import styles from "./Button.module.scss";
+
+export type ButtonColor =
+  | "primary"
+  | "secondary"
+  | "success"
+  | "warning"
+  | "neutral"
+  | "support"
+  | "inverse";
```

Comment:
i went back and forth on this a lot. the issues i ran into when trying to come up with a good representation for different colors is that not all of our outlined variations have filled counterparts and its not particularly obvious how to handle invalid states. i did think about just dumping all of the variations into a single property, something like 

`ButtonColorStyles = “primary_filled” | “primary_outlined” | “secondary_filled”, | “secondary_outlined” ..etc` 

and im still open to it, but it does deviate quite a bit from how we’ve been doing it. 

with the way i implemented it here, we do still have some invalid states, but handling that under this structure would mean detecting a fallback which im not sure how friendly that is to the dx.

curious if you have thoughts and im open to ideas.

Reply:
i like what you decided on

if there are invalid variant combinations, we can log a warning: https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Icon/Icon.tsx#L60